server/plugin.go

```diff
@@ -97,281 +85,85 @@ func (p *Plugin) OnActivate() error {
 		return errors.Wrap(appErr, "couldn't set profile image")
 	}
 
-	p.zoomClient = zoom.NewClient(config.ZoomAPIURL, config.APIKey, config.APISecret)
+	p.jwtClient = zoom.NewJWTClient(p.getZoomAPIURL(), config.APIKey, config.APISecret)
 
 	return nil
 }
 
-func (p *Plugin) getSiteURL() (string, error) {
-	siteURLRef := p.API.GetConfig().ServiceSettings.SiteURL
-	if siteURLRef == nil || *siteURLRef == "" {
-		return "", errors.New("error fetching siteUrl")
+// registerSiteURL fetches the site URL and sets it in the plugin object.
+func (p *Plugin) registerSiteURL() error {
+	siteURL := p.API.GetConfig().ServiceSettings.SiteURL
+	if siteURL == nil || *siteURL == "" {
+		return errors.New("could not fetch siteURL")
 	}
 
-	return *siteURLRef, nil
+	p.siteURL = *siteURL
+	return nil
 }
 
-func (p *Plugin) getOAuthConfig() (*oauth2.Config, error) {
+// getActiveClient returns an OAuth Zoom client if available, otherwise it returns the API client.
+func (p *Plugin) getActiveClient(user *model.User) (zoom.Client, error) {
 	config := p.getConfiguration()
 
-	clientID := config.OAuthClientID
-	clientSecret := config.OAuthClientSecret
-	zoomURL := config.ZoomURL
-	if zoomURL == "" {
-		zoomURL = zoomDefaultURL
+	if !config.EnableOAuth {
+		return p.jwtClient, nil
 	}
 
-	authURL := fmt.Sprintf("%v/oauth/authorize", zoomURL)
-	tokenURL := fmt.Sprintf("%v/oauth/token", zoomURL)
+	info, err := p.fetchOAuthUserInfo(zoomTokenKey, user.Id)
```

Comment:
What do you think changing this `zoomTokenKey` to something like `zoomUserByMMID`, and also update `zoomTokenByZoomIDKye` to something like `zoomUserByZoomID`?

It would be only a change in the variables names, since a change in the content would be a breaking change.